swarm/storage/pyramid.go

```diff
@@ -155,140 +155,140 @@ type PyramidChunker struct {
 	chunkLevel  [][]*TreeEntry
 }
 
-func NewPyramidSplitter(params *PyramidSplitterParams) (self *PyramidChunker) {
-	self = &PyramidChunker{}
-	self.reader = params.reader
-	self.hashSize = params.hashSize
-	self.branches = params.chunkSize / self.hashSize
-	self.chunkSize = self.hashSize * self.branches
-	self.putter = params.putter
-	self.getter = params.getter
-	self.key = params.addr
-	self.workerCount = 0
-	self.jobC = make(chan *chunkJob, 2*ChunkProcessors)
-	self.wg = &sync.WaitGroup{}
-	self.errC = make(chan error)
-	self.quitC = make(chan bool)
-	self.rootKey = make([]byte, self.hashSize)
-	self.chunkLevel = make([][]*TreeEntry, self.branches)
+func NewPyramidSplitter(params *PyramidSplitterParams) (pc *PyramidChunker) {
```

Comment:
The same suggestion as for NewTreeJoiner, even `pc` is using its fields in setting other fields, they are all originated from params.